chalk-ir/src/lib.rs

```diff
@@ -2461,13 +2753,26 @@ impl<'i, I: Interner> Folder<'i, I> for &SubstFolder<'i, I> {
 /// additionally be solved.
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Fold, Visit, HasInterner)]
 pub struct ConstrainedSubst<I: Interner> {
-    pub subst: Substitution<I>, /* NB: The `is_trivial` routine relies on the fact that `subst` is folded first. */
+    /// The substitution that is being constrained.
+    ///
+    /// NB: The `is_trivial` routine relies on the fact that `subst` is folded first.
+    pub subst: Substitution<I>, /*  */
```

Comment:
Extra comments now

Reply:
Heh, good catch. I glanced right over that the first 5 times I read your comment :laughing: